tests/unit/components/DocumentationTopic/DocumentationNav/VersionToggle.spec.js

```diff
@@ -0,0 +1,215 @@
+/**
+ * This source file is part of the Swift.org open source project
+ *
+ * Copyright (c) 2022 Apple Inc. and the Swift project authors
+ * Licensed under Apache License v2.0 with Runtime Library Exception
+ *
+ * See https://swift.org/LICENSE.txt for license information
+ * See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import { RouterLinkStub, shallowMount } from '@vue/test-utils';
+import DocumentationTopicStore from 'docc-render/stores/DocumentationTopicStore';
+import VersionToggle
+  from 'docc-render/components/DocumentationTopic/DocumentationNav/VersionToggle.vue';
+import { flushPromises } from '../../../../../test-utils';
+
+const { NavMenuItemBase } = VersionToggle.components;
+
+describe('VersionToggle`', () => {
+  let wrapper;
+
+  const propsData = {
+    versionList: ['V3, V2, V1'],
+  };
+
+  const mocks = {
+    $route: {
+      path: '/documentation/foo',
+    },
+    $router: {
+      push: jest.fn(),
+    },
+  };
+
+  const provide = {
+    store: {
+      setPreferredVersion: jest.fn(),
+    },
+  };
+
+  const stubs = {
+    'router-link': RouterLinkStub,
+  };
+
+  const createWrapper = (props = propsData, mocksData = mocks) => (
+    shallowMount(VersionToggle, {
+      mocks: mocksData,
+      propsData: props,
+      provide,
+      stubs,
+    })
+  );
+
+  beforeEach(() => {
+    wrapper = createWrapper();
+    localStorage.clear();
+    DocumentationTopicStore.reset();
+  });
+
+  it('renders `NavMenuItemBase` at the root', () => {
+    const menuItemBase = wrapper.find(NavMenuItemBase);
+    expect(menuItemBase.exists()).toBe(true);
+    expect(menuItemBase.classes()).toContain('nav-menu-setting');
+    expect(menuItemBase.classes()).toContain('version-container');
+  });
+
+  it('does not render the version toggle component when there are no versions', () => {
```

Comment:
Here your test preparation was bad. You had just one version, so the `hasMultipleVersions` check was being falsy.

```suggestion
  it('renders a single span if there is only one version', () => {
```